docs/migration/2211/2211-installation.md

```diff
@@ -1 +1,110 @@
+```diff
+  {
+    "projects": {
+      ...
+      "(my app name)": {
+        ...
+        "architect": {
+          ...
+          "build": {
+            ...
+            "configurations": {
+              "production": {
+                ...
+              },
+              "development": {
+                ...
+              },
++             "noSsr": {
++               "ssr": false,
++               "prerender": false
++             }
+            },
+```
+2. In the section `projects › (my app name) › architect › serve › configurations` append a string `,noSsr` (including comma) to the value of the properties `"buildTarget"` - both for 2 sections `"production"` and `"development"`:
+```diff
+  {
+    "projects": {
+      ...
+      "(my app name)": {
+        ...
+        "architect": {
+          ...
+          "serve": {
+            ...
+            "configurations": {
+              "production": {
+-               "buildTarget": "test-ng17:build:production"
++               "buildTarget": "test-ng17:build:production,noSsr"
+              },
+              "development": {
+-               "buildTarget": "test-ng17:build:development"
++               "buildTarget": "test-ng17:build:development,noSsr"
+              }
+            },
+```
+
+**Why**: Since Angular 17, in freshly created new apps (not migrated ones from Angular 16!) Angular runs SSR and Prerendering by default a part of `ng serve` and `ng build` commands. Unfortunately Angular 17 also introduced a new bug: *`server.ts` file is _totally ignored_ during the `ng serve` command (see Angular issue https://github.com/angular/angular-cli/issues/26323)*. Spartacus to work correctly, requires `server.ts` file to be included in the compilation and execution. Therefore, for Spartacus to work, as a workaround, you have to disable SSR and Prerendering for `ng serve` in your `angular.json`.
+
+### Disable Prerendering for `ng build`
+In `angular.json` of newly created app make the following change:
+In the section `projects › (my app name) › architect › build › options` remove the line `prerender: true`:
+
+```diff
+  {
+    "projects": {
+      ...
+      "(my app name)": {
+        ...
+        "architect": {
+          ...
+          "build": {
+            ...
+            "options": {
+              "production": {
+                ...
+-               "prerender": "true"
+              }
+            },
+```
+
+**Why**: Since Angular 17, in freshly created new apps (not migrated ones from Angular 16!) Angular runs Prerendering by default a part of `ng build` commands (and transitively also for `ng serve`). But Spartacus, for Prerendering to work correctly, requires a custom Node Environment Variable `SERVER_REQUEST_ORIGIN` to be defined. Otherwise Spartacus Prerendering would fail with an error about missing `SERVER_REQUEST_ORIGIN`. So as a workaround, you have to disable Prerendering for the `ng build`, to be able to build app at least for CSR or SSR. 
```

Comment:
```suggestion
**Why**: Since Angular 17, in freshly created new apps (not migrated ones from Angular 16!) Angular runs Prerendering by default as a part of `ng build` commands (and transitively also for `ng serve`). But Spartacus, for Prerendering to work correctly, requires a custom Node Environment Variable `SERVER_REQUEST_ORIGIN` to be defined. Otherwise, Spartacus Prerendering would fail with an error about missing `SERVER_REQUEST_ORIGIN`. So as a workaround, you have to disable Prerendering for the `ng build`, to be able to build the app at least for CSR or SSR. 
```